sky/utils/subprocess_utils.py

```diff
@@ -293,3 +293,23 @@ def kill_process_daemon(process_pid: int) -> None:
         # Disable input
         stdin=subprocess.DEVNULL,
     )
+
+
+def maybe_parallelize_cloud_operation(
+        func: Callable,
+        clouds: List[Any],
+        num_threads: Optional[int] = None) -> List[Any]:
+    """Apply a function to a list of clouds,
+    with parallelism if there is more than one cloud.
+    """
+    count = len(clouds)
+    if count == 0:
+        return []
+    # Short-circuit in single cloud setup.
+    if count == 1:
+        return [func(clouds[0])]
```

Comment:
How about we have the `run_in_parallel` to automatically choose whether to short-circuit a thread pool based on the length of the input? In this case, we can avoid a new API, which also benefits the existing code.

Reply:
Good call, thanks!